tests/ui/lint/invalid_value.stderr

```diff
@@ -34,8 +34,7 @@ LL |         let _val: Wrap<&'static T> = mem::zeroed();
    |                                      this code causes undefined behavior when executed
    |                                      help: use `MaybeUninit<T>` instead, and only call `assume_init` after initialization is done
    |
-   = note: `Wrap<&T>` must be non-null
```

Comment:
This diff is surprising, why did that line disappear?

Reply:
So, I've looked into it, and what's happening is that the lint is calling `layout_of(Wrap<&T>)` to get extra information on the type (`rustc_lint/src/builtins.rs:2484`). This now fails, because the exact layout (including niches) now depends on `T`.

Reply:
Ah, and `T` is generic here. Makes sense.

As long as the lint still fires, seems okay to me!